src/main/java/eu/dissco/core/digitalspecimenprocessor/service/ProcessingService.java

```diff
@@ -130,6 +130,41 @@ private ProcessResult processSpecimens(Set<DigitalSpecimenEvent> events) {
     }
   }
 
+  /*
+  We need to remove the EntityRelationshipDate from the comparison.
+  We cannot control the equals in the EntityRelationship as it is generated
+  Therefore we first store the EntityRelationshipDate in a map and set both to null
+  When the objects are not equal we reinsert the data into the EntityRelationship
+  */
+  private boolean isEquals(DigitalSpecimenWrapper currentDigitalSpecimen,
+      DigitalSpecimenWrapper digitalSpecimen) {
+    var entityRelationships = digitalSpecimen.attributes().getEntityRelationship();
+    digitalSpecimen.attributes().setEntityRelationship(
+        entityRelationships.stream().map(this::deepcopyEntityRelationship).toList());
+    ignoreTimestampEntityRelationship(currentDigitalSpecimen.attributes().getEntityRelationship());
+    if (currentDigitalSpecimen.equals(digitalSpecimen)) {
```

Comment:
do you not need to set the entity relationship back in both cases?

Reply:
No, if they are equal (which if we run harvesting it most often is) it is an unnecessary action. We will only update the last checked for the specimen. However, it might be unnecessary it might make it unclear. Maybe just do it anyway so we end with a full specimen instead of a tampered with, just for maintainability/extensibilities purpose. Will change it